common/culture/traditions/00_ritual_traditions.txt

```diff
@@ -639,14 +647,23 @@ tradition_sorcerous_metallurgy = {
 	}
 
 	is_shown = {
-	}
-
-	can_pick = {
-		scope:character = {
-			faith = {
-				has_doctrine = doctrine_witchcraft_accepted
+		OR = {
+			scope:character = {
+				faith = {
+					has_doctrine = doctrine_witchcraft_accepted
+				}
+				any_vassal_or_below = {
```

Comment:
Prob too CPU heavy check.

Reply:
I'm glad you caught that! I didn't realise that `any_vassal_or_below` checked all vassals, not just direct ones. How is this? 

```
				any_vassal = {
					count >= 3
					OR = {
						has_trait = magic_good_noninheritable
						has_trait = magic_good_1
						has_trait = magic_good_2
						has_trait = magic_good_3
					}
				}
```